nox/_options.py

```diff
@@ -12,23 +12,202 @@
+        "-s",
+        "-e",
+        "--sessions",
+        group="primary",
+        noxfile=True,
+        merge_func=functools.partial(_sessions_and_keywords_merge_func, "sessions"),
+        nargs="*",
+        default=_sessions_default,
+        help="Which sessions to run, by default, all sessions will run.",
+    ),
+    _option_set.Option(
+        "keywords",
+        "-k",
+        "--keywords",
+        noxfile=True,
+        merge_func=functools.partial(_sessions_and_keywords_merge_func, "keywords"),
+        help="Only run sessions that match the given expression.",
+    ),
+    _option_set.Option(
+        "posargs",
+        "posargs",
+        group="primary",
+        nargs=argparse.REMAINDER,
+        help="Arguments following -- that are passed through to the session(s).",
+        finalizer_func=_posargs_finalizer,
+    ),
+    *_option_set.make_flag_pair(
+        "reuse_existing_virtualenvs",
+        ["-r", "--reuse-existing-virtualenvs"],
+        ["--no-reuse-existing-virtualenvs"],
+        group="secondary",
+        help="Re-use existing virtualenvs instead of recreating them.",
+    ),
+    _option_set.Option(
+        "noxfile",
+        "-f",
+        "--noxfile",
+        group="secondary",
+        default="noxfile.py",
+        help="Location of the Python file containing nox sessions.",
+    ),
+    _option_set.Option(
+        "envdir",
+        "--envdir",
+        noxfile=True,
+        merge_func=_envdir_merge_func,
+        group="secondary",
+        help="Directory where nox will store virtualenvs, this is .nox by default.",
+    ),
+    *_option_set.make_flag_pair(
+        "stop_on_first_error",
+        ["-x", "--stop-on-first-error"],
+        ["--no-stop-on-first-error"],
+        group="secondary",
+        help="Stop after the first error.",
+    ),
+    *_option_set.make_flag_pair(
+        "error_on_missing_interpreters",
+        ["--error-on-missing-interpreters"],
+        ["--no-error-on-missing-interpreters"],
+        group="secondary",
+        help="Error instead of skipping sessions if an interpreter can not be located.",
+    ),
+    *_option_set.make_flag_pair(
+        "error_on_external_run",
+        ["--error-on-external-run"],
+        ["--no-error-on-external-run"],
+        group="secondary",
+        help="Error if run() is used to execute a program that isn't installed in a session's virtualenv.",
+    ),
+    _option_set.Option(
+        "install_only",
+        "--install-only",
+        group="secondary",
+        action="store_true",
+        help="Skip session.run invocations in the Noxfile.",
+    ),
+    _option_set.Option(
+        "report",
+        "--report",
+        group="secondary",
+        noxfile=True,
+        help="Output a report of all sessions to the given filename.",
+    ),
+    _option_set.Option(
+        "nocolor",
+        "--nocolor",
+        group="secondary",
+        default=lambda: "NO_COLOR" in os.environ or not sys.stderr.isatty(),
```

Comment:
`lambda` not really necessary here. Just write a function? (I try not to use `lambda`, in honor of [Guido][1].)

[1]: https://www.artima.com/weblogs/viewpost.jsp?thread=98196

Reply:
I removed this entirely as part of adding the mutually exclusivity of the color options.